system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -520,11 +605,75 @@ cpr::Response ElasticStackLogAccess::performESQuery(const LogAccessConditions &
 bool ElasticStackLogAccess::fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format)
 {
     cpr::Response esresp = performESQuery(options);
-    processESJsonResp(esresp, returnbuf, format);
+    processESSearchJsonResp(esresp, returnbuf, format);
 
     return true;
 }
 
+class ELASTICSTACKLOGACCESS_API ElasticStackLogStream : public CInterfaceOf<IRemoteLogAccessStream>
+{
+public:
+    virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead, unsigned & totalRecs) override
+    {
+        Json::Value res;
+        recsRead = 0;
+        totalRecs = 0;
+
+        if (m_esSroller.next(res))
+        {
+
+            if (!res["total"]["value"].empty())
+            {
+                totalRecs = res["total"]["value"].asUInt();
+            }
+
+            if (!res["hits"].empty())
+            {
+                recsRead = res["hits"].size();
+                std::ostringstream sout;
+                m_jsonWriter->write(res, &sout);
```

Comment:
design: This process is a bit convoluted, and was hard for me to follow.

I think the sequence is
* The value comes back in a Json object structure
* Written that value as json
* Read the json in as an IPropertyTree
* iterate elements in the IPropertyTree
* convert to json/csv/xml as required.

That may be the best way to code it, but it would be useful to have a comment clarifying that (I originally thought the json writer here was formatting the final output).  It is probably good for reusing code, but it is potentially quite inefficient - especially if these logs get large.

Reply:
optimally we would serialize directly from JSON::Value to json/csv/xml. This change would be somewhat involved. How critical do you think it is to optimize this now?

Reply:
@ghalliday  Jira to address optimization: https://track.hpccsystems.com/browse/HPCC-27423